File: 백준/Bronze/2798. 블랙잭/블랙잭.rs
use std::{
    io::{stdin, Read},
    str::FromStr,
};

fn main() {
    let m = input_line_splitted_as::<i32>()[1];
    let nums = input_all_splitted_as::<i32>();

    let mut closest_sum = 0;
    let n = nums.len();

    for i in 0..n - 2 {
        for j in i + 1..n - 1 {
            for k in j + 1..n {
                let sum = nums[i] + nums[j] + nums[k];
                if sum <= m && sum > closest_sum {
                    closest_sum = sum;
                }
            }
        }
    }

    println!("{closest_sum}");
}

fn input_line_splitted_as<T>() -> Vec<T>
where
    T: Eq + Ord + Copy + FromStr,
    <T as FromStr>::Err: std::fmt::Debug,
{
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();

    buf.trim()
        .split_ascii_whitespace()
        .map(|s| s.parse().unwrap())
        .collect::<Vec<_>>()
}

fn input_all_splitted_as<T>() -> Vec<T>
where
    T: FromStr,
    <T as FromStr>::Err: std::fmt::Debug,
{
    let mut buf = String::new();
    stdin().read_to_string(&mut buf).unwrap();

    buf.trim()
        .split_ascii_whitespace()
        .map(|s| s.parse().unwrap())
        .collect::<Vec<_>>()
}

File: 백준/Bronze/2798. 블랙잭/블랙잭.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int n, m;
  cin >> n >> m;

  vector<int> nums(n);

  for (int i = 0; i < n; i++) {
    cin >> nums[i];
  }
  sort(nums.begin(), nums.end());

  int closest_sum = 0;

  for (int i = 0; i < n - 2; i++) {
    for (int j = i + 1; j < n - 1; j++) {
      for (int k = j + 1; k < n; k++) {
        int sum = nums[i] + nums[j] + nums[k];
        if (sum > closest_sum && sum <= m) {
          closest_sum = sum;
        }
      }
    }
  }

  cout << closest_sum;

  return 0;
}